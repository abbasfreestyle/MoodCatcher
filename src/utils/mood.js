export const minRange = 1;

export const maxRange = 7;

export const moodPercentage = mood => (mood - minRange) / (maxRange - minRange);
