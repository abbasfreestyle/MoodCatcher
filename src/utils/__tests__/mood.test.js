import { minRange, maxRange, moodPercentage } from '../mood';

describe('Utils > Mood', () => {
  describe('Snapshots.', () => {
    test('should match the minRange snapshot', () => {
      expect(minRange).toMatchSnapshot();
    });
    test('should match the maxRange snapshot', () => {
      expect(maxRange).toMatchSnapshot();
    });
  });

  describe('Behavior.', () => {
    test('should output 0 if the min range is provided.', () => {
      expect(moodPercentage(minRange)).toEqual(0);
    });
    test('should output 1 if the max range is provided.', () => {
      expect(moodPercentage(maxRange)).toEqual(1);
    });

    test('should output 0.5 if the middle of the range is provided.', () => {
      expect(moodPercentage((minRange + maxRange) / 2)).toEqual(0.5);
    });
  });
});
