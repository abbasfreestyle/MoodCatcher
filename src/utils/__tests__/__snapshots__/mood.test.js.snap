// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils > Mood Snapshots. should match the maxRange snapshot 1`] = `7`;

exports[`Utils > Mood Snapshots. should match the minRange snapshot 1`] = `1`;
